### ISSUE TITLE
[Question] Jobs getting stuck in Redis before reaching maximum number of attempts

### ISSUE THREAD

**Issue by alireza8101** [OptimalBits/bull#2571]
I'm encountering an issue where some jobs are getting stuck in Redis before reaching the maximum number of attempts in my Node.js application. I've configured the queue with the following options:
```
const queueOptions = {
  defaultJobOptions: {
    attempts: 100,
    backoff: 3000,
    timeout: 24 * 60 * 60 * 1000,
    removeOnComplete: true,
    removeOnFail: true,
  },
};
```
This should ensure that jobs are removed from the queue after reaching maximum attempts, as long as the job has been completed or failed. However, I've noticed some jobs getting stuck in Redis before reaching maximum attempts. For example, a job that has been attempted just 28 times but remains in Redis for three days and is not being removed from Redis by Bull, it looks like Bull can not identify this job.

I've tried using the `empty()`, and `clean()` methods on the Bull queue, but I still see stuck jobs in Redis that have been there for several days. I've also checked the Redis logs and metrics but have no apparent issues or errors.

I'm wondering if there may be an issue with the Bull library or Redis configuration causing jobs to get stuck in Redis and not properly removed. Any help or guidance would be greatly appreciated.

**Comment by manast** [comment]
First of all, in which state is the given job? "stuck" does not exist as a state in Bull, so it must be either in waiting, delayed, completed or failed.

**Comment by alireza8101** [comment]
maybe it could help?
```
> HGETALL bull:myQueue:99414
 1) "name"
 2) "myQueue"
 3) "stacktrace"
 4) "[\"..."]"
 5) "failedReason"
 6) "Unable to load data"
 7) "delay"
 8) "0"
 9) "priority"
10) "0"
11) "data"
12) "{...}"
13) "timestamp"
14) "1679418290165"
15) "opts"
16) "{\"attempts\":100,\"backoff\":{\"type\":\"fixed\",\"delay\":3000},\"timeout\":86400000,\"removeOnComplete\":true,\"removeOnFail\":true,\"delay\":0,\"timestamp\":1679418290165}"
17) "attemptsMade"
18) "51"
19) "processedOn"
20) "1679418440983"
```
